File: java/cours/src/session2/td/agregation/Nat.java
package session2.td.agregation;

public interface Nat extends EtatNaturel<Nat>, hierarchie.SemiAnneauUnitaireEuclidien<Nat>, session1.td.FabriqueNaturels<Nat>{
	Nat creerNatAvecEtat(EtatNaturelPur etat);
	EtatNaturelPur etat();
}

File: typescript/session2/td/agregation/Nat.ts
import { EtatNaturel } from "./EtatNaturel";
import { FabriqueNat } from "../../../session1/td/naturels";
import { SemiAnneauUnitaireEuclidien } from "../../../structuresAlgebriques/hierarchie";
import { EtatNaturelPur } from "./EtatNaturelPur";

export interface Nat extends EtatNaturel<Nat>, FabriqueNat<Nat>, SemiAnneauUnitaireEuclidien<Nat> {
    creerNatAvecEtat(etat : EtatNaturelPur) : Nat;
    etat() : EtatNaturelPur;
    estEgal(x : Nat) : boolean;
    representation() : string;
}